### ISSUE TITLE
Deployment via CLI fails when adding package "@genkit-ai/firebase": "^0.9.12",

### ISSUE THREAD

**Issue by AttilaJF** [firebase/genkit#1523]
**Describe the bug**
When trying to deploy project using CLI command firebase deploy.
the following error report emerges:

Build failed with status: FAILURE and message: npm error code ERESOLVE
npm error ERESOLVE could not resolve
npm error
npm error While resolving: @genkit-ai/firebase@0.9.12
npm error Found: firebase-admin@11.11.1
npm error node_modules/firebase-admin
npm error   firebase-admin@"^11.11.1" from the root project
npm error   peer firebase-admin@"^11.0.1 || ^12.0.0" from firebase-frameworks@0.11.6
npm error   node_modules/firebase-frameworks
npm error     firebase-frameworks@"^0.11.0" from the root project
npm error   1 more (firebase-functions)
npm error
npm error Could not resolve dependency:
npm error peer firebase-admin@">=12.2" from @genkit-ai/firebase@0.9.12
npm error node_modules/@genkit-ai/firebase
npm error   @genkit-ai/firebase@"^0.9.12" from the root project
npm error
npm error Conflicting peer dependency: firebase-admin@13.0.1
npm error node_modules/firebase-admin
npm error   peer firebase-admin@">=12.2" from @genkit-ai/firebase@0.9.12
npm error   node_modules/@genkit-ai/firebase
npm error     @genkit-ai/firebase@"^0.9.12" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.

The dependency issue also occurs when trying different versions.
**To Reproduce**
1. start nextjs project with package.json below:
{
  "private": true,
  "scripts": {
    "dev": "next dev --turbopack",
    "build": "next build",
    "start": "next start"
  },
  "dependencies": {
    "@genkit-ai/dev-local-vectorstore": "^0.9.12",
    "@genkit-ai/googleai": "^0.9.12",
    "@genkit-ai/vertexai": "^0.9.12",
    "classnames": "^2.5.1",
    "date-fns": "^3.6.0",
    "genkit": "^0.9.12",
    "gray-matter": "^4.0.3",
    "llm-chunk": "^0.0.1",
    "next": "15.0.2",
    "pdf-parse": "^1.1.1",
    "react": "19.0.0-rc-02c0e824-20241028",
    "react-dom": "19.0.0-rc-02c0e824-20241028",
    "remark": "^15.0.1",
    "remark-html": "^16.0.1"
  },
  "devDependencies": {
    "@types/node": "^20.14.8",
    "@types/pdf-parse": "^1.1.4",
    "@types/react": "npm:types-react@19.0.0-rc.1",
    "@types/react-dom": "npm:types-react-dom@19.0.0-rc.1",
    "autoprefixer": "^10.4.19",
    "postcss": "^8.4.38",
    "tailwindcss": "^3.4.4",
    "typescript": "^5.5.2"
  }
}

3. initialize firebase project and use command firebase deploy
4. The project deploys without any issues
5. install package: "@genkit-ai/firebase": "^0.9.12",
6. deploy project with firebase deploy
7. The above error report is printed

**Expected behavior**
The project should deploy without dependency issues. Locally running the project does not shows issues.

**Desktop (please complete the following information):**
 - Device: [MacBook Pro
 - OS: [macOS 14.4.1 (23E224)]
 - Browser [Chrome, Firefox, Safari,]
 

**Additional context**
A RAG project was built and runs fine locally. Deployment is not possible using "@genkit-ai/firebase" for retrieval.
import { defineFirestoreRetriever } from "@genkit-ai/firebase";
// export const faqRetriever = defineFirestoreRetriever(ai, {
//   name: "faqRetriever",
//   firestore: getFirestore(app),
//   collection: "studentFaq",
//   contentField: "text",
//   vectorField: "embedding",
//   embedder: textEmbedding004, // Import from '@genkit-ai/googleai' or '@genkit-ai/vertexai'
//   distanceMeasure: "COSINE", // "EUCLIDEAN", "DOT_PRODUCT", or "COSINE" (default)
// });

**Comment by Naryalin** [comment]
Same here with Astro
```
npm error code ERESOLVE
npm error ERESOLVE unable to resolve dependency tree
npm error
npm error While resolving: veritheca@0.0.1
npm error Found: firebase-admin@13.0.2
npm error node_modules/firebase-admin
npm error   firebase-admin@"^13.0.2" from the root project
npm error
npm error Could not resolve dependency:
npm error peer firebase-admin@"^11.0.1 || ^12.0.0" from firebase-frameworks@0.11.6
npm error node_modules/firebase-frameworks
npm error   firebase-frameworks@"^0.11.0" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
```

**Comment by AttilaJF** [comment]
@Naryalin , have you tried adding an 
.npmrc file to the root directory,
 and in that file add the line: legacy-peer-deps=true

**Comment by Naryalin** [comment]
> [@Naryalin](https://github.com/Naryalin) , have you tried adding an .npmrc file to the root directory, and in that file add the line: legacy-peer-deps=true

No, for now I’ve just downgraded the `firebase-admin` version since I’m still in the development phase. I’ll try your approach unless a version of firebase-frameworks compatible with firebase-admin@13.0.2 is released—thanks for the suggestion
